src/compiler.ts

```diff
@@ -1933,6 +1937,7 @@ export class Compiler extends DiagnosticEmitter {
     var tableBase = this.options.tableBase;
     if (!tableBase) tableBase = 1; // leave first elem blank
     index = tableBase + functionTable.length;
+    assert(u32(index) < u32(2147483648));
```

Comment:
```suggestion
    assert(index >= 0);
```

Reply:
Why would that work? Index is a u32 so I'd expect it to always be `>= 0`

Reply:
I suggest leave this as is as more clear check or leave some comment about signed overflow